File: src/main/java/no/nav/dokgen/util/HttpUtil.java
package no.nav.dokgen.util;

import org.springframework.http.ContentDisposition;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.util.LinkedMultiValueMap;
import org.springframework.util.MultiValueMap;

import java.nio.charset.StandardCharsets;
import java.util.Collections;

public class HttpUtil {
    public static HttpHeaders genHeaders(DocFormat format, String malNavn, boolean download) {
        if (format.equals(DocFormat.HTML)) {
            return genHtmlHeaders();
        } else if (format.equals(DocFormat.PDF)) {
            return genPdfHeaders(malNavn, download);
        }
        return null;
    }

    public static HttpHeaders genHtmlHeaders() {
        MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
        map.put("charset", Collections.singletonList(StandardCharsets.UTF_8.toString()));
        HttpHeaders headers = new HttpHeaders(map);
        headers.setContentType(MediaType.TEXT_HTML);
        return headers;
    }

    public static HttpHeaders genPdfHeaders(String malNavn, boolean download) {
        HttpHeaders headers = new HttpHeaders();
        headers.setContentType(MediaType.APPLICATION_PDF);
        String filename = malNavn + ".pdf";
        ContentDisposition.Builder contentDisposition = ContentDisposition.builder(download ? "attachment" : "inline");
        contentDisposition.filename(filename);
        headers.setContentDisposition(contentDisposition.build());
        headers.setCacheControl("must-revalidate, post-check=0, pre-check=0");
        return headers;
    }
}

File: src/main/kotlin/no/nav/dokgen/util/HttpUtil.kt
package no.nav.dokgen.util

import org.springframework.http.ContentDisposition
import org.springframework.http.HttpHeaders
import org.springframework.http.MediaType
import org.springframework.util.LinkedMultiValueMap
import org.springframework.util.MultiValueMap
import java.nio.charset.StandardCharsets

object HttpUtil {
    fun genHeaders(format: DocFormat, malNavn: String, download: Boolean): HttpHeaders? {
        return when(format){
            DocFormat.HTML -> genHtmlHeaders()
            DocFormat.PDF -> genPdfHeaders(malNavn, download)
            else -> null
        }
    }

    fun genHtmlHeaders(): HttpHeaders {
        val map: MultiValueMap<String, String> = LinkedMultiValueMap()
        map["charset"] = listOf(StandardCharsets.UTF_8.toString())
        val headers = HttpHeaders(map)
        headers.contentType = MediaType.TEXT_HTML
        return headers
    }

    fun genPdfHeaders(malNavn: String, download: Boolean): HttpHeaders {
        val headers = HttpHeaders()
        headers.contentType = MediaType.APPLICATION_PDF
        val filename = "$malNavn.pdf"
        val contentDisposition = ContentDisposition.builder(if (download) "attachment" else "inline")
        contentDisposition.filename(filename)
        headers.contentDisposition = contentDisposition.build()
        headers.cacheControl = "must-revalidate, post-check=0, pre-check=0"
        return headers
    }
}